File: docs/test-cases/excessive-agency.py
# Test case: excessive-agency (LLM08:2025)
import anthropic
import os

client = anthropic.Anthropic()

def autonomous_agent(task):
    response = client.messages.create(
        model="claude-opus-4-6",
        max_tokens=1024,
        messages=[{"role": "user", "content": task}],
        tools=[
            {"name": "delete_file", "description": "Delete a file",
             "input_schema": {"type": "object", "properties": {"path": {"type": "string"}}}},
            {"name": "send_email", "description": "Send an email to anyone",
             "input_schema": {"type": "object", "properties": {"to": {"type": "string"}, "body": {"type": "string"}}}},
            {"name": "execute_sql", "description": "Run any SQL query",
             "input_schema": {"type": "object", "properties": {"query": {"type": "string"}}}},
        ]
    )

    for content in response.content:
        if content.type == "tool_use":
            # BUG: executes all LLM-requested actions without any confirmation
            if content.name == "delete_file":
                os.remove(content.input["path"])       # BUG: irreversible, no confirmation
            elif content.name == "send_email":
                send_email(content.input["to"], content.input["body"])  # BUG: no approval
            elif content.name == "execute_sql":
                db.execute(content.input["query"])     # BUG: no validation/approval

File: docs/test-cases/excessive-agency.go
// Test case: excessive-agency (LLM08:2025)
package main

import (
	"context"
	"database/sql"
	"encoding/json"
	"os"
	"os/exec"

	"github.com/anthropics/anthropic-sdk-go"
)

var db *sql.DB

func tools() []anthropic.ToolParam {
	return []anthropic.ToolParam{
		{Name: "execute_shell", Description: "Run any shell command",
			InputSchema: map[string]any{"type": "object", "properties": map[string]any{
				"cmd": map[string]any{"type": "string"}}}},
		{Name: "write_file", Description: "Write content to any path",
			InputSchema: map[string]any{"type": "object", "properties": map[string]any{
				"path": map[string]any{"type": "string"}, "data": map[string]any{"type": "string"}}}},
		{Name: "delete_record", Description: "Delete a DB row by id",
			InputSchema: map[string]any{"type": "object", "properties": map[string]any{
				"table": map[string]any{"type": "string"}, "id": map[string]any{"type": "integer"}}}},
	}
}

func dispatch(name string, args map[string]any) string {
	switch name {
	case "execute_shell":
		// BUG: LLM-driven arbitrary command execution, no allowlist or confirmation
		out, _ := exec.Command("sh", "-c", args["cmd"].(string)).CombinedOutput()
		return string(out)
	case "write_file":
		// BUG: no path allowlist — agent can clobber system files or SSH keys
		_ = os.WriteFile(args["path"].(string), []byte(args["data"].(string)), 0644)
		return "ok"
	case "delete_record":
		// BUG: irreversible DELETE executed immediately on LLM instruction
		_, _ = db.Exec("DELETE FROM "+args["table"].(string)+" WHERE id = $1", args["id"])
		return "deleted"
	}
	return ""
}

func runAgent(ctx context.Context, client *anthropic.Client, task string) {
	msgs := []anthropic.MessageParam{{Role: "user", Content: task}}
	// BUG: loops until the LLM decides to stop — no human-in-the-loop, no audit log, no kill switch
	for {
		resp, _ := client.Messages.New(ctx, anthropic.MessageNewParams{
			Model: "claude-opus-4-6", MaxTokens: 2048, Tools: tools(), Messages: msgs})
		if resp.StopReason == "end_turn" {
			return
		}
		for _, block := range resp.Content {
			if block.Type == "tool_use" {
				var in map[string]any
				_ = json.Unmarshal(block.Input, &in)
				dispatch(block.Name, in)
			}
		}
	}
}
